cpu/atmega_common/periph/rtt.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2016 Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ *               2018 Acutam Automation, LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/*
+ * @ingroup cpu_atmega_common
+ * @ingroup drivers_periph_rtt
+ * @{
+ *
+ * @file
+ * @brief       Low-level ATmega RTT driver implementation
+ *
+ * @author      Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * In order to safely sleep when using the RTT:
+ * 1. Disable interrupts
+ * 2. Write to one of the asynch registers (e.g. TCCR2A)
+ * 3. Wait for ASSR register's busy flags to clear
+ * 4. Re-enable interrupts
+ * 5. Sleep before interrupt re-enable takes effect
+ *
+ * @}
+ */
+
+#include <avr/interrupt.h>
+
+#include "cpu.h"
+#include "irq.h"
+#include "periph/rtt.h"
+#include "periph_conf.h"
+#include "thread.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* guard file in case no RTT device is defined */
+#if RTT_NUMOF
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if MODULE_PERIPH_RTC
+extern void atmega_rtc_incr(void);
+#endif
+
+static inline void _asynch_wait(void);
+
+typedef struct {
+    uint16_t ext_cnt;           /* Counter to make 8-bit timer 24-bit */
+    uint16_t ext_comp;          /* Extend compare to 24-bits */
+    rtt_cb_t alarm_cb;          /* callback called from RTT alarm */
+    void *alarm_arg;            /* argument passed to the callback */
+    rtt_cb_t overflow_cb;       /* callback called when RTT overflows */
+    void *overflow_arg;         /* argument passed to the callback */
+} rtt_state_t;
+
+static volatile rtt_state_t rtt_state;
```

Comment:
Is `volatile` correct here? If so, please add comments explaining it.

Reply:
Comment added. Also, see post below.